test/integration/instance.test.js

```diff
@@ -512,6 +512,136 @@ describe(Support.getTestDialectTeaser('Instance'), function() {
         });
       });
     });
+
+    it('should set an association to null after deletion, 1-1', function() {
+      var Shoe = this.sequelize.define('Shoe', { brand: DataTypes.STRING })
+        , Player = this.sequelize.define('Player', { name: DataTypes.STRING });
+
+      Player.hasOne(Shoe);
+      Shoe.belongsTo(Player);
+
+      return Player.sync({force: true}).then(function() {
```

Comment:
This is pretty waterfally - could be rewritten as:

```js


return this.sequelize.sync({force: true}).then(function() {
  return Shoe.create({
    brand: 'the brand',
    Player: {
      name: 'the player'
    }
  }, {include: [Player]});
}).then(function(shoe) {
  return Player.findOne({
    where: { id: shoe.Player.id },
    include: [Shoe]
  }).then(function(lePlayer) {
    expect(lePlayer.Shoe).not.to.be.null;
    return lePlayer.Shoe.destroy().return(lePlayer);
  }).then(function(lePlayer) {
    return lePlayer.reload();
  }).then(function(lePlayer) {
    expect(lePlayer.Shoe).to.be.null;
  });
});
```

I know its just copy-pasta from some of the methods above, but we like to keep a less waterfall approach in new code :)